src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/ScheduleController.java

```diff
@@ -95,19 +97,50 @@ public String add( @PathVariable("scheduleid") Long scheduleid,
         primary.setQuarter(lecture_quarter);
         primary.setScheduleid(scheduleid);
         logger.info("primary = " + primary); 
-        scheduleItemRepository.save(primary);
+        // scheduleItemRepository.save(primary);
 
         return "redirect:/schedule";
     }
 
     @PostMapping("/schedule/create")
-    public String add_schedule(String sname, Model model, OAuth2AuthenticationToken token) {
+    public String add_schedule(Model model, OAuth2AuthenticationToken token) {
         Schedule newschedule = new Schedule();
         String uid = token.getPrincipal().getAttributes().get("sub").toString();
         newschedule.setUid(uid);
-        newschedule.setSchedulename(sname);
+        newschedule.setSchedulename("New Schedule");
         scheduleRepository.save(newschedule);
         return "redirect:/schedule";
     }
 
+    @GetMapping("/schedule/{scheduleid}")
+    public String viewSchedule(Model model, OAuth2AuthenticationToken token, @PathVariable("scheduleid") Long scheduleid) {
+        
+        logger.info("Inside /schedule controller method ScheduleController#viewschedule");
+        logger.info("model=" + model + " token=" + token);
+
+        if (token!=null) {
+            String uid = token.getPrincipal().getAttributes().get("sub").toString();
+            logger.info("uid="+uid);
+            logger.info("scheduleItemRepository="+scheduleItemRepository);
+            List<Schedule> myschedules = scheduleRepository.findByUid(uid);// get all schedule ids by uid
+            // get courses by each scheduleid to a list
+            Iterable<ScheduleItem> myclasses = scheduleItemRepository.findByScheduleid(scheduleid);
+            Schedule schedule = new Schedule();
+            for (int i = 0; i < myschedules.size(); i++) {
+                if (myschedules.get(i).getScheduleid() == scheduleid) {
+                    schedule = (myschedules.get(i));
+                }
+            }
```

Comment:
In my opinion, this for loop seems completely unnecessary if we already have the scheduleid of the schedule we want to display. Since we already have the ID number, we could use  that to directly grab that schedule, instead of having to loop through every schedule to check if the ID number matches.

Reply:
I believe List type in java doesn’t have a search by element function. I think this is necessary unless there is a way to do it with Repositories which we don’t know enough about. :/

Reply:
If you already have the scheduleid, you wouldn't even need to get a list of schedules in the first place... you could just use that to find it in the repositories similar to how you're using a user's uid to find their schedules.

Reply:
I'll try that.